webapp/vue.md

```diff
@@ -0,0 +1,53 @@
+# Vue – Code Guidelines
+
+## We use single-file components
+
+Each component lives in a single file, containing:
+- its `template` (the DOM structure)
+- its `script` (including `props`, `data` and `methods` among other things)
+- its `style` (defining the look of the component)
+
+See the [Vue.js docs](https://vuejs.org/v2/guide/single-file-components.html) for more details.
+
+Placed in the same folder are also:
+- the test file (e.g. `MyComponent.spec.js`)
+- the storybook file (e.g. `MyComponent.story.js`)
+
+## We use typed props
+
+Vue.js allows us to define component props either as strings or as objects (with `type` and `default` or `required` values). Always go for the second option!
+
+Also: only (and always!) define a `default` for props that are _not required_.
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-31T11:11:57Z" title="Thursday, October 31st 2019, 12:11:57 pm +01:00">Oct 31, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,53 @@
+# Vue – Code Guidelines
+
+## We use single-file components
+
+Each component lives in a single file, containing:
+- its `template` (the DOM structure)
+- its `script` (including `props`, `data` and `methods` among other things)
+- its `style` (defining the look of the component)
+
+See the [Vue.js docs](https://vuejs.org/v2/guide/single-file-components.html) for more details.
+
+Placed in the same folder are also:
+- the test file (e.g. `MyComponent.spec.js`)
+- the storybook file (e.g. `MyComponent.story.js`)
+
+## We use typed props
+
+Vue.js allows us to define component props either as strings or as objects with `type`, `default` and `required` values. Always go for the second option!
+
+Also: define defaults _only and always_ for non-required props.
```


I feel like we did the opposite in our pairing session the other day, oder?
we removed the `required: true`, and gave it `default: 'alert'`, no?

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-11-04T09:16:45Z" title="Monday, November 4th 2019, 10:16:45 am +01:00">Nov 4, 2019</time>_

---


Yep, that's what I was trying to say in this sentence. :wink: I'll rephrase it!